chat-android/src/main/java/com/ably/chat/RoomOptions.kt

```diff
@@ -1,113 +1,187 @@
      * in order to subscribe to presence.
      * @defaultValue true
      */
-    val subscribe: Boolean = true,
-)
+    public val subscribe: Boolean
+}
 
 /**
  * Represents the typing options for a chat room.
  */
-public data class TypingOptions(
+public interface TypingOptions {
     /**
      * The timeout for typing events in milliseconds. If typing.start() is not called for this amount of time, a stop
      * typing event will be fired, resulting in the user being removed from the currently typing set.
      * @defaultValue 5000
      */
-    val timeoutMs: Long = 5000,
-)
+    public val timeoutMs: Long
+}
 
 /**
  * Represents the reactions options for a chat room.
  *
  * Note: This class is currently empty but allows for future extensions
  * while maintaining backward compatibility.
  */
-public class RoomReactionsOptions {
-    override fun equals(other: Any?): Boolean = other is RoomReactionsOptions
-    override fun hashCode(): Int = javaClass.hashCode()
-}
+public interface RoomReactionsOptions
 
 /**
  * Represents the occupancy options for a chat room.
  *
  * Note: This class is currently empty but allows for future extensions
  * while maintaining backward compatibility.
  */
-public class OccupancyOptions {
-    override fun equals(other: Any?): Boolean = other is OccupancyOptions
-    override fun hashCode(): Int = javaClass.hashCode()
+public interface OccupancyOptions
+
+@ChatDsl
+public class MutableRoomOptions : RoomOptions {
+    override var presence: MutablePresenceOptions? = null
+    override var typing: MutableTypingOptions? = null
+    override var reactions: MutableRoomReactionsOptions? = null
+    override var occupancy: MutableOccupancyOptions? = null
+}
+
+@ChatDsl
+public class MutablePresenceOptions : PresenceOptions {
+    override var enter: Boolean = true
+    override var subscribe: Boolean = true
+}
+
+@ChatDsl
+public class MutableTypingOptions : TypingOptions {
+    override var timeoutMs: Long = 5000
+}
+
+@ChatDsl
+public class MutableRoomReactionsOptions : RoomReactionsOptions
+
+@ChatDsl
+public class MutableOccupancyOptions : OccupancyOptions
+
+public fun buildRoomOptions(init: MutableRoomOptions.() -> Unit = {}): RoomOptions =
+    MutableRoomOptions().apply(init).asEquatable()
+
+public fun MutableRoomOptions.presence(init: MutablePresenceOptions.() -> Unit = {}) {
+    this.presence = MutablePresenceOptions().apply(init)
 }
 
+public fun MutableRoomOptions.typing(init: MutableTypingOptions.() -> Unit = {}) {
+    this.typing = MutableTypingOptions().apply(init)
+}
+
+public fun MutableRoomOptions.reactions(init: MutableRoomReactionsOptions.() -> Unit = {}) {
+    this.reactions = MutableRoomReactionsOptions().apply(init)
+}
+
+public fun MutableRoomOptions.occupancy(init: MutableOccupancyOptions.() -> Unit = {}) {
+    this.occupancy = MutableOccupancyOptions().apply(init)
+}
+
+internal data class EquatableRoomOptions(
```

Comment:
Why don't call this `DefaultRoomOptions` to keep naming consistency 🤔

Reply:
Because purpose of this class to create object with `equals` defined correctly, so we can compare `RoomOptions`

Reply:
Resolving based on https://github.com/ably/ably-chat-kotlin/pull/119/files#r2010279280